cymetric/graphs.py

```diff
@@ -31,35 +41,74 @@ def flow_graph(evaler, senders=(), receivers=(), commodities=(), nucs=(),
     start : first timestep to consider, start included
     stop : last timestep to consider, stop included
     """
-    tools.raise_no_graphviz('Unable to generate flow graph!', HAVE_GRAPHVIZ)
+    tools.raise_no_graphviz("Unable to generate flow graph!", HAVE_GRAPHVIZ)
 
-    df = transactions_nuc(
-        evaler, senders, receivers, commodities, nucs)
+    df = transactions_nuc(evaler, senders, receivers, commodities, nucs)
 
     if start is not None:
-        df = df.loc[(df['Time'] >= start)]
+        df = df.loc[(df["Time"] >= start)]
     if stop is not None:
-        df = df.loc[(df['Time'] <= stop)]
+        df = df.loc[(df["Time"] <= stop)]
 
-    group_end = ['ReceiverPrototype', 'SenderPrototype', 'Commodity']
-    group_start = group_end + ['Mass']
+    group_end = ["ReceiverPrototype", "SenderPrototype", "Commodity"]
+    group_start = group_end + ["Mass"]
     df = df[group_start].groupby(group_end).sum()
     df.reset_index(inplace=True)
 
-    agents_ = evaler.eval('AgentEntry')['Prototype'].tolist()
+    agents_ = evaler.eval("AgentEntry")
 
-    dot = Digraph('G')
+    dot = Digraph("G", strict=True)
 
-    for agent in agents_:
-        dot.node(agent)
+    # start by constructing region subgraphs
+    regions = agents_[agents_["Kind"] == "Region"]
+    for i, row in regions.iterrows():
+        region_id = row["AgentId"]
+        region_prototype = row["Prototype"]
+        institutions = agents_[
+            (agents_["ParentId"] == region_id) & (agents_["Kind"] == "Inst")
+        ]
```

Comment:
In line with using `Kind` to filter for `Region`, we can also use `Inst`to ensure that we are filtering the appropriate agents.